### ISSUE TITLE
🐛[BUG]: Using BFGS as optimizer,but TensorBoard only output a loss value

### ISSUE THREAD

**Issue by YouqiongLiu** [NVIDIA/physicsnemo-sym#202]
### Version

modulus-sym:1.5.0

### Describe the issue

How to output loss value per iter when using BFGS optimizer?
and the TensorBoard only output a loss value

### Relevant log output

[21:00:24] - lbfgs optimizer selected. Setting max_steps to 0
[21:00:24] - [step:          0] lbfgs optimization in running
[21:19:36] - lbfgs optimization completed after 20000 steps
[21:19:36] - [step:          0] record constraint batch time:  7.847e-02s
[21:19:37] - [step:          0] record validators time:  3.937e-01s
[21:19:37] - [step:          0] record inferencers time:  2.740e-01s
[21:19:39] - [step:          0] saved checkpoint to outputs/ldc_2d_bfgs
[21:19:39] - [step:          0] loss:  3.858e-03
[21:19:39] - [step:          0] reached maximum training steps, finished training!

**Comment by ktangsali** [comment]
The iteratioins of the BFGS optimizer are not exposed to the Tensorboard loggers and hence this is an expected behavior.
